embedded-graphics/src/primitives/polyline.rs

```diff
@@ -0,0 +1,402 @@
+            line_iter: self.primitive.points(),
+        }
+    }
+}
+
+/// Pixel iterator for each pixel in the line
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct StyledPolylineIterator<'a, C>
+where
+    C: PixelColor,
+{
+    stroke_width: u32,
+    stroke_color: Option<C>,
+    line_iter: Points<'a>,
+}
+
+impl<'a, C: PixelColor> Iterator for StyledPolylineIterator<'a, C> {
+    type Item = Pixel<C>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        // Break if stroke width is zero
+        if self.stroke_width == 0 {
+            return None;
+        }
+
+        // Return none if stroke color is none
+        let stroke_color = self.stroke_color?;
+
+        self.line_iter
+            .next()
+            .map(|point| Pixel(point, stroke_color))
+    }
+}
+
+impl<'a, C: 'a> Drawable<C> for &Styled<Polyline<'a>, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    fn draw<D: DrawTarget<C>>(self, display: &mut D) -> Result<(), D::Error> {
+        display.draw_iter(self.into_iter())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{
+        mock_display::MockDisplay,
+        pixelcolor::{BinaryColor, Rgb565},
+        prelude::*,
+    };
+
+    // A "heartbeat" shaped polyline
+    const HEARTBEAT: [Point; 10] = [
+        Point::new(10, 64),
+        Point::new(50, 64),
+        Point::new(60, 44),
+        Point::new(70, 64),
+        Point::new(80, 64),
+        Point::new(90, 74),
+        Point::new(100, 10),
+        Point::new(110, 84),
+        Point::new(120, 64),
+        Point::new(300, 64),
+    ];
+
+    // Smaller test pattern for mock display
+    const SMALL: [Point; 4] = [
+        Point::new(2, 5),
+        Point::new(5, 2),
+        Point::new(10, 5),
+        Point::new(15, 2),
+    ];
+
+    #[test]
+    fn positive_dimensions() {
+        let polyline = Polyline::new(&HEARTBEAT);
+
+        let bb = polyline.bounding_box();
+
+        assert_eq!(
+            bb,
+            Rectangle::with_corners(Point::new(10, 10), Point::new(300, 84))
+        );
+    }
+
+    #[test]
+    fn negative_dimensions() {
+        let mut negative: [Point; 10] = [Point::zero(); 10];
+
+        for (i, v) in HEARTBEAT.iter().enumerate() {
+            negative[i] = *v - Point::new(100, 100);
+        }
+
+        let polyline = Polyline::new(&negative);
+
+        let bb = polyline.bounding_box();
+
+        assert_eq!(
+            bb,
+            Rectangle::with_corners(Point::new(-90, -90), Point::new(200, -16))
+        );
+    }
+
+    #[test]
+    fn transformed_dimensions() {
+        let polyline = Polyline::new(&HEARTBEAT).translate(Point::new(-100, -100));
+
+        let bb = polyline.bounding_box();
+
+        assert_eq!(
+            bb,
+            Rectangle::with_corners(Point::new(-90, -90), Point::new(200, -16))
+        );
+    }
+
+    #[test]
+    fn one_point() {
+        let _polyline = Polyline::new(&[Point::zero()]);
```

Comment:
You could also add a test if the points iterator works correctly for this case.

Reply:
Added!